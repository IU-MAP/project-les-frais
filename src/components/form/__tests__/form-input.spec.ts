import { mount } from '@vue/test-utils';
import FormInput from '../form-input.vue';

describe('FormInput', () => {
  const setupWrapper = (props: any) => mount(FormInput, {
    ...props,
  });
  describe('when it has no label', () => {
    const component = setupWrapper({ noLabel: true });
    it('matches the snapshot', () => {
      expect(component).toMatchSnapshot();
    });

    it('emits focusin and focusout events', async () => {
      await component.find('input').trigger('focusin');
      expect(component.emitted()).toHaveProperty('focusin');
      await component.find('input').trigger('focusout');
      expect(component.emitted()).toHaveProperty('focusout');
    });
  });

  describe('when it has label', () => {
    const component = setupWrapper({ labelText: 'label' });
    it('matches the snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('when it has the basic HTML attributes', () => {
    const component = setupWrapper({
      type: 'email',
      autocomplete: 'on',
      placeholder: 'google@innopolis.university',
      required: true,
      disabled: true,
    });
    it('matches the snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
