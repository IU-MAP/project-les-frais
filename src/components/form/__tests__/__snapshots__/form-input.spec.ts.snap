// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput when it has label matches the snapshot 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "__emits": null,
      "__props": Array [
        Object {},
        Array [],
      ],
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <div
        class="form-input"
      >
        <label
          class="form-input_label"
        >
          
          
          
        </label>
        <div
          class="form-input_wrapper"
        >
          
          
          <input
            autocomplete="off"
            class="form-input_input"
            placeholder=""
            type="text"
          />
          
          
        </div>
        <!--v-if-->
      </div>
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {
        "transition": Object {
          "name": "transition",
          "props": undefined,
          "render": [Function],
        },
        "transition-group": Object {
          "name": "transition-group",
          "props": undefined,
          "render": [Function],
        },
      },
      "config": Object {
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isCustomElement": [Function],
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "mixins": Array [
        Object {
          "__emits": null,
          "__props": Array [],
          "beforeCreate": [Function],
        },
      ],
      "provides": Object {},
      "reload": [Function],
    },
    "_props": null,
    "_uid": 1,
    "component": [Function],
    "config": Object {
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isCustomElement": [Function],
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.0.11",
  },
  "__setProps": [Function],
  "componentVM": Object {
    "autocomplete": "off",
    "disabled": false,
    "error": "",
    "hasOwnProperty": [Function],
    "id": null,
    "input": [Function],
    "inputAttrs": Object {},
    "inputDelay": 0,
    "labelText": "",
    "mask": "",
    "noLabel": false,
    "placeholder": "",
    "required": false,
    "type": "text",
    "val": null,
    "value": null,
  },
  "rootVM": Object {},
  "wrapperElement": <div
    class="form-input"
  >
    <label
      class="form-input_label"
    >
      
      
      
    </label>
    <div
      class="form-input_wrapper"
    >
      
      
      <input
        autocomplete="off"
        class="form-input_input"
        placeholder=""
        type="text"
      />
      
      
    </div>
    <!--v-if-->
  </div>,
}
`;

exports[`FormInput when it has no label matches the snapshot 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "__emits": null,
      "__props": Array [
        Object {},
        Array [],
      ],
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <div
        class="form-input"
      >
        <label
          class="form-input_label"
        >
          
          
          
        </label>
        <div
          class="form-input_wrapper"
        >
          
          
          <input
            autocomplete="off"
            class="form-input_input"
            placeholder=""
            type="text"
          />
          
          
        </div>
        <!--v-if-->
      </div>
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {
        "transition": Object {
          "name": "transition",
          "props": undefined,
          "render": [Function],
        },
        "transition-group": Object {
          "name": "transition-group",
          "props": undefined,
          "render": [Function],
        },
      },
      "config": Object {
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isCustomElement": [Function],
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "mixins": Array [
        Object {
          "__emits": null,
          "__props": Array [],
          "beforeCreate": [Function],
        },
      ],
      "provides": Object {},
      "reload": [Function],
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": Object {
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isCustomElement": [Function],
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.0.11",
  },
  "__setProps": [Function],
  "componentVM": Object {
    "autocomplete": "off",
    "disabled": false,
    "error": "",
    "hasOwnProperty": [Function],
    "id": null,
    "input": [Function],
    "inputAttrs": Object {},
    "inputDelay": 0,
    "labelText": "",
    "mask": "",
    "noLabel": false,
    "placeholder": "",
    "required": false,
    "type": "text",
    "val": null,
    "value": null,
  },
  "rootVM": Object {},
  "wrapperElement": <div
    class="form-input"
  >
    <label
      class="form-input_label"
    >
      
      
      
    </label>
    <div
      class="form-input_wrapper"
    >
      
      
      <input
        autocomplete="off"
        class="form-input_input"
        placeholder=""
        type="text"
      />
      
      
    </div>
    <!--v-if-->
  </div>,
}
`;

exports[`FormInput when it has the basic HTML attributes matches the snapshot 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "__emits": null,
      "__props": Array [
        Object {},
        Array [],
      ],
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <div
        class="form-input"
      >
        <label
          class="form-input_label"
        >
          
          
          
        </label>
        <div
          class="form-input_wrapper"
        >
          
          
          <input
            autocomplete="off"
            class="form-input_input"
            placeholder=""
            type="text"
          />
          
          
        </div>
        <!--v-if-->
      </div>
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {
        "transition": Object {
          "name": "transition",
          "props": undefined,
          "render": [Function],
        },
        "transition-group": Object {
          "name": "transition-group",
          "props": undefined,
          "render": [Function],
        },
      },
      "config": Object {
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isCustomElement": [Function],
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "mixins": Array [
        Object {
          "__emits": null,
          "__props": Array [],
          "beforeCreate": [Function],
        },
      ],
      "provides": Object {},
      "reload": [Function],
    },
    "_props": null,
    "_uid": 2,
    "component": [Function],
    "config": Object {
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isCustomElement": [Function],
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.0.11",
  },
  "__setProps": [Function],
  "componentVM": Object {
    "autocomplete": "off",
    "disabled": false,
    "error": "",
    "hasOwnProperty": [Function],
    "id": null,
    "input": [Function],
    "inputAttrs": Object {},
    "inputDelay": 0,
    "labelText": "",
    "mask": "",
    "noLabel": false,
    "placeholder": "",
    "required": false,
    "type": "text",
    "val": null,
    "value": null,
  },
  "rootVM": Object {},
  "wrapperElement": <div
    class="form-input"
  >
    <label
      class="form-input_label"
    >
      
      
      
    </label>
    <div
      class="form-input_wrapper"
    >
      
      
      <input
        autocomplete="off"
        class="form-input_input"
        placeholder=""
        type="text"
      />
      
      
    </div>
    <!--v-if-->
  </div>,
}
`;
